File: sdks/java/fireweave-sdk/src/main/java/ai/fireweave/sdk/domain/Mode.java
package ai.fireweave.sdk.domain;

/**
 * SDK runtime mode (spec/modes.md). An SDK instance runs in exactly one mode, fixed at
 * initialisation; the mode selects the adapter and nothing downstream branches on it again.
 *
 * <p>{@code mode} is required and never inferred: {@code InitOptions.mode()} is nullable
 * precisely so a caller (or a config file half-migrated from remote to local) can construct an
 * options value with no mode at all, which {@code Validation.validateInitOptions} rejects as
 * {@code Configuration} — "mode absent" in spec/modes.md's initialisation-validation table.
 * "mode unrecognised" (the table's other half of that row) has no Java analogue: an enum-typed
 * field cannot hold a value outside {@link #LOCAL}/{@link #REMOTE} in the first place.
 */
public enum Mode {
    LOCAL,
    REMOTE
}

File: sdks/go/domain/mode.go
package domain

// Mode is the SDK runtime mode (spec/modes.md). An SDK instance runs in
// exactly one mode, fixed at initialisation; the mode selects the adapter
// and nothing downstream branches on it again.
//
// Mode is a plain string type rather than a closed enum deliberately: it is
// required and never inferred (spec/modes.md "mode is required and never
// inferred"), and ValidateInitOptions must be able to reject BOTH "absent"
// (the zero value "") and "unrecognised" (any string other than the two
// canonical values) as Configuration — a closed enum would make the latter
// unrepresentable, matching a caller-facing string option (e.g. read from a
// half-migrated config file) more faithfully than an enum would.
type Mode string

const (
	ModeLocal  Mode = "local"
	ModeRemote Mode = "remote"
)
